src/memory/wxorx.rs

```diff
@@ -5,29 +5,28 @@ use super::{
 };
 
 use bytes::Bytes;
-use std::marker::PhantomData;
 
-pub struct WXorXMemory<R: Register, M: Memory<R>> {
+pub struct WXorXMemory<M: Memory> {
     inner: M,
-    _inner: PhantomData<R>,
```

Comment:
I am curious why we add a PhantomData a before, and why it makes sense to delete it for now.

Reply:
when using generics type in `Memory` trait, we have to annotate the types in each implementation.